Change Model::row_data to return an Option<T> (#873)
sixtyfpsui/sixtyfps: api/sixtyfps-cpp/docs/cpp_migration.md

## Migrating from Version 0.1.x to 0.2.0

> In 0.2.0 we have increased the minimum version of C++. You need to have a C++ compiler installed that supports C++ 20 or newer.

Reviewer: ```suggestion
In version 0.2.0 we have increased the minimum version of C++. You need to have a C++ compiler installed that supports C++ 20 or newer.
```

sixtyfps::Value args[] = { SharedString("Hello"), 42. };
instance->invoke_callback("foo", args);
```

#### Models

> `Model::row_data` will now return a `std::optional<ModelData>` and can thus be used with indices that are out of bounds.

Reviewer: ```suggestion
`Model::row_data` returns now a `std::optional<ModelData>` and can thus be used with indices that are out of bounds.
```

Old code:

```cpp
std::get<2>(another_model->row_data(2)), 336.);
```

New code:

> ```cpp
> std::get<2>(*another_model->row_data(2)), 336.); // Note the extra *. You should check whether the returned value `has_value()` before accessing it

Reviewer: ```suggestion
```cpp
// `another_model` is a model that contains floats.
std::optional<float> entry = another_model->row_data(2);
if (entry.has_value()) {
    do_something(*entry);
} else {
    // row index 2 is out of bounds
}
```

sixtyfps::Value args[] = { SharedString("Hello"), 42. };
instance->invoke_callback("foo", args);
```

#### Models

`Model::row_data` returns now a `std::optional<ModelData>` and can thus be used with indices that are out of bounds.

This also means that `Model`s must handle invalid indices and may not crash when a invalid index is passed in.

Old code:

```cpp

> another_model->row_data(2);

Reviewer: I think it would be a good idea to use the return value in a way that is similar to the "new" code.

```suggestion
float value = another_model->row_data(2);
do_something(value);
```
